### PR TITLE
Prevent null date-time strings from deserializing to 'now'

### PR DESCRIPTION
**Pull request by childish-sambino**

### CODE REVIEW

File: Twilio/Deserialize.php [hunk 1]
```diff
@@ -7,16 +7,19 @@ class Deserialize {
...
      */
     public static function dateTime($s) {
         try {
-            return new \DateTime($s, new \DateTimeZone('UTC'));
+            if ($s) {
+                return new \DateTime($s, new \DateTimeZone('UTC'));
+            }
         } catch (\Exception $e) {
-            return $s;
+            // no-op
```
Comment (krantitalluri) [inline Twilio/Deserialize.php]: don't we need to log this as standard practice?

Comment (childish-sambino) [reply Twilio/Deserialize.php]: We pretty much don't do any logging in the helper libs. I don't like silent-failing _anywhere_, but that's the current status quo.

Comment (thinkingserious) [reply Twilio/Deserialize.php]: How about capturing this as issues across our repos?

Comment (childish-sambino) [reply Twilio/Deserialize.php]: Added.

**Comment by tmcnicholls** [comment]
I think this should have been flagged as a BC break in the release notes for the SDK 5.36.0.

This change caused us an issue in production where we make the assumption that the dateSent property of a sent or accepted SMS will always be set. This PR now means that it returns a null if the SMS was only accepted onto the message queue, and not yet sent.

Do you follow semver in your releases? We weren't expecting a patch version release to have any breaking changes.

**Comment by childish-sambino** [comment]
@tmcnicholls This lib uses a modified semver strategy (doc [here](https://github.com/twilio/twilio-php/blob/master/VERSIONS.md)).

I can understand that this change caused an issue, but it is really a bug fix. In your case, `dateSent` was really `null` and the library was misrepresenting that to be 'now'.

Since there was a breaking change elsewhere, the minor version rolled so in the end this is really a release notes issue. I'm okay with updating the changelog to indicate such, but the version would still have been bumped the same way given our versioning strategy.

**Pull request merged**